### ISSUE TITLE
Logic operators expected to be functions

### ISSUE THREAD

**Issue by DeiRex75** [pantor/inja#154]
If you use 'and', 'or', 'not' in combination with parenthesis on the right expression, a parser error is thrown. Saying for example 'unknown function not'.

```
nlohmann::json data;
inja::render(" {{ not true }} ", data);  
=> false
```

```
nlohmann::json data;
inja::render(" {{ not (true) }} ", data);  
=> [inja.exception.parser_error] unknown function not
```

Similar:
{{ true or (true) }}  
{{ true or (true or true) }}

**Comment by DeiRex75** [comment]
I took a look myself and it seems to be an order problem.
#155

**Comment by pantor** [comment]
Fixed by #155.
